### PR TITLE
[15.0][MIG] sale_order_revision

### PR DESCRIPTION
**Pull request by leemannd**
Superseeds https://github.com/OCA/sale-workflow/pull/1931

### CODE REVIEW

**Comment by CRogos** [comment]
I'll close mine, too. Same state as yours ;)
#2224

**Comment by leemannd** [comment]
@CRogos A review is welcome
@simahawk A merge is welcome

**Review by cyrilmanuel — APPROVED** [review]
Thx you for this job !

**Review by cyrilmanuel — APPROVED** [review]

**Comment by leemannd** [comment]
Hello @OCA/crm-sales-marketing-maintainers A review and eventually a merge are welcome

File: sale_order_revision/__init__.py [hunk 1]
```diff
@@ -0,0 +1,10 @@
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+from . import models
+
+
+def populate_unrevisioned_name(cr, registry):
```
Comment (grindtildeath) [inline sale_order_revision/__init__.py:5]: This should be moved into `hooks.py` and the function renamed to `post_init_hook`. You can then add `from.hooks import post_init_hook` here. cf what's done in https://github.com/OCA/maintainer-tools/tree/master/template/module

File: sale_order_revision/__init__.py [hunk 2]
```diff
@@ -0,0 +1,10 @@
...
+    cr.execute(
+        "UPDATE sale_order "
+        "SET unrevisioned_name = name "
+        "WHERE unrevisioned_name is NULL"
```
Comment (grindtildeath) [inline sale_order_revision/__init__.py:9]:
```suggestion
        "WHERE unrevisioned_name is NULL;"
```

**Review by grindtildeath — APPROVED** [review]
Not blocking but would be good. Pre approving anyway.

**Review by CRogos — CHANGES_REQUESTED** [review]
Could you have a look on this error in the log? Is the test executed?
image

You should also squash the commits.

File: sale_order_revision/__init__.py (hunk 1 above)
Comment (simahawk) [reply sale_order_revision/__init__.py:5]: yes pls

File: rest-requirements.txt [hunk 3]
```diff
@@ -0,0 +1 @@
+odoo_test_helper
```
Comment (simahawk) [inline rest-requirements.txt:1]: this file is wrongly named

File: oca_dependencies.txt [hunk 4]
```diff
@@ -0,0 +1 @@
+server-ux
```
Comment (simahawk) [inline oca_dependencies.txt:1]: this file is useles

File: sale_order_revision/tests/test_sale_order_revision.py [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+from odoo.tests import Form
+
+from odoo.addons.base_revision.tests import test_base_revision
+
+
+class TestSaleOrderRevision(test_base_revision.TestBaseRevision):
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+        cls.revision_model = cls.env["sale.order"]
```
Comment (simahawk) [inline sale_order_revision/tests/test_sale_order_revision.py:13]: tracking_disable pls

**Review by simahawk — CHANGES_REQUESTED** [review]

File: oca_dependencies.txt (hunk 4 above)
Comment (leemannd) [reply oca_dependencies.txt:1]: Hum, from what has it been replaced?

Comment (simahawk) [reply oca_dependencies.txt:1]: Is useless on:

1. old repo has "PIP" install mode instead of "OCA"
2. all new repos using GH actions

**Comment by simahawk** [comment]
> Could you have a look on this error in the log? Is the test executed? image
> 
> You should also squash the commits.

this is the problem https://github.com/OCA/sale-workflow/pull/2275#discussion_r1035588075

**Comment by simahawk** [comment]
#2381

**Pull request closed without merging**